### DISCUSSION TITLE
How to edit or delete a Kubernetes cluster?

### DISCUSSION THREAD

**Discussion by checor** [gravitational/teleport#21328] [Q&A] [answered]
Hello,

I'm currently configuring Teleport Cloud edition. I have added an Kubernetes cluster to my cluster, using a Helm chart.

However, I need to change the kubernetes user and role that it's used to perform kubectl actions.

On the Web App, I only see a "Connect" button. Not a way to delete or edit a Kubernetes cluster. Is there a way to edit or delete this connection?

![Paste](https://user-images.githubusercontent.com/3594530/217056812-93ff40b8-f0dd-4a95-96c0-8fbffc20973d.png)

**Comment by webvictim** [comment, marked as answer]
The `teleport-kube-agent` chart that you installed to your cluster using Helm will continue "heartbeating" this cluster as long as the Helm deployment is running.

If you run `helm ls -aA` you should see all the Helm releases in your cluster. You can then use `helm uninstall -n <namespace> <release-name>` to uninstall the `teleport-kube-agent` release. The Kubernetes cluster will disappear from the Teleport web UI around 10-15 minutes after the release is uninstalled as the pod will stop heartbeating.

**Comment by ronaldmiranda** [comment]
@webvictim  on the new versions, im having dangling kubernetes on my OSS teleport, how can i remove it?

**Reply by webvictim** [reply]
You can `tctl rm kube_server/<name>` to manually remove a Kubernetes cluster.
